lib/broadway/server.ex

```diff
@@ -103,23 +103,33 @@ defmodule Broadway.Server do
 
     n_producers = producer_config[:stages]
 
-    names =
+    names_and_specs =
       for index <- 1..n_producers do
-        producer_name(name_prefix(broadway_name), index)
-      end
+        name = producer_name(name_prefix(broadway_name), index)
+
+        # Inject the topology index only if the producer-specific options are a keyword list.
+        producer_config =
```

Comment:
Let's break the `producer_config` into `{module, config}` and all `Producer.start_link(module, config, opts)`.

Reply:
`producer_config` is "one level higher" than that. It's

```
[
  module: {BroadwayRabbitMQ.Producer, [...]},
  stages: 10
]
```

We're just calling `Producer.start_link(producer_config, opts)` here. If we do what you said, we'd also have to pass in the `:transformer` in the options so we'd either have to put that in the `opts` (together with GenServer-specific options) or add it as an arg. What do you prefer? I'm inclined to leave it like this for now :)

Reply:
I see. So let’s pass the index on start_link and we put it altogether in the producer. I just want to avoid multiple packing/unpacking. Wdyt?

Reply:
Sure, makes sense. Done and pushed :)